src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import { getFilteredPeople } from '../../utils/getFilteredPeople';
+import { getSortedPeople } from '../../utils/getSortedPeople';
+import { SortBy } from '../../types/SortBy';
+import { SearchLink } from '../SearchLink';
+
+type Props = {
+  people: Person[],
+  personSlug: string,
+};
+
+export const PeopleTable: React.FC<Props> = React.memo(
+  ({ people, personSlug }) => {
+    const [searchParams] = useSearchParams();
+    const sort = searchParams.get('sort') as keyof Person;
+    const isReversed = searchParams.get('order') === 'desc';
+    const query = searchParams.get('query');
+    const sexFilter = searchParams.get('sex');
+    const centuries = searchParams.getAll('centuries');
+
+    const visiblePeople = useMemo(() => (
+      getFilteredPeople(people, query, sexFilter, centuries)
+    ), [people, query, sexFilter, centuries]);
+
+    const sortedPeople = useMemo(() => (
+      getSortedPeople(visiblePeople, sort, isReversed)
+    ), [visiblePeople, sort, isReversed]);
+
+    if (!sortedPeople.length) {
+      return <p>There are no people matching the current search criteria</p>;
+    }
+
+    const updateSearchParams = (sortBy: SortBy) => {
+      if (!sort || sort !== sortBy) {
+        return ({
+          sort: sortBy,
+          order: null,
+        });
+      }
+
+      if (sort && !isReversed) {
+        return ({
+          sort: sortBy,
+          order: 'desc',
+        });
+      }
+
+      return ({
+        sort: null,
+        order: null,
+      });
+    };
+
+    const handleSortClass = (sortBy: SortBy) => {
+      if (sort === sortBy && !isReversed) {
+        return 'fas fa-sort-up';
+      }
+
+      if (sort === sortBy && isReversed) {
+        return 'fas fa-sort-down';
+      }
+
+      return 'fas fa-sort';
+    };
```

Comment:
As a suggestion, we can use here also `classnames`.
```suggestion
    const handleSortClass = classnames('fas', {
        'fa-sort': sort !== sortBy,
        'fa-sort-up': sort === sortBy && !isReversed,
        'fa-sort-down': sort === sortBy && isReversed,
    });
```

Reply:
agree, check @Avichez comment